### PR TITLE
🌱 CI: add annotation with prowjob name and id to ipclaims

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Adds the prowjob name and id as annotation to ipclaims to help identifying where leaks of IPs come from.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -27,8 +27,8 @@ source "${REPO_ROOT}/hack/ensure-kubectl.sh"
 on_exit() {
   # release IPClaim
   echo "Releasing IP claims"
-  kubectl --kubeconfig="${KUBECONFIG}" delete "$(append_api_group ipclaim)" "${IPCLAIM_NAME}" || true
-  kubectl --kubeconfig="${KUBECONFIG}" delete "$(append_api_group ipclaim)" "${WORKLOAD_IPCLAIM_NAME}" || true
+  kubectl --kubeconfig="${KUBECONFIG}" delete "$(append_api_group ipclaim)" "${CONTROL_PLANE_IPCLAIM_NAME}" || true
+  kubectl --kubeconfig="${KUBECONFIG}" delete "$(append_api_group ipclaim)" "${WORKLOAD_PLANE_IPCLAIM_NAME}" || true
```
Comment (sbueringer) [inline hack/e2e.sh:31]: WORKLOAD_PLANE_IPCLAIM_NAME sounds a bit strange :)

File: Makefile [hunk 2]
```diff
@@ -466,7 +471,7 @@ e2e-image: ## Build the e2e manager image
 
 .PHONY: e2e
 e2e: e2e-image generate-e2e-templates
-e2e: $(GINKGO) $(KUSTOMIZE) $(KIND) $(GOVC) ## Run e2e tests
+e2e: $(GINKGO) $(KUSTOMIZE) $(KIND) $(GOVC) $(ENVSUBST) ## Run e2e tests
```
Comment (sbueringer) [inline Makefile:474]: Do we actually have to build envsubst here at this point?

File: hack/e2e.sh [hunk 3]
```diff
@@ -101,20 +101,29 @@ function kubectl_get_jsonpath() {
...
+CONTROL_PLANE_ENDPOINT_IP=$(claim_ip "${CONTROL_PLANE_IPCLAIM_NAME}")
 export CONTROL_PLANE_ENDPOINT_IP
 echo "Acquired Control Plane IP: $CONTROL_PLANE_ENDPOINT_IP"
 
 # Retrieve an IP to be used for the workload cluster in v1a3/v1a4 -> v1b1 upgrade tests
-WORKLOAD_IPCLAIM_NAME="workload-ip-claim-$(openssl rand -hex 20)"
-sed "s/IPCLAIM_NAME/${WORKLOAD_IPCLAIM_NAME}/" "${REPO_ROOT}/hack/ipclaim-template.yaml" | kubectl --kubeconfig=${KUBECONFIG} create -f -
-WORKLOAD_IPADDRESS_NAME=$(kubectl_get_jsonpath ipclaim "${WORKLOAD_IPCLAIM_NAME}" '{@.status.address.name}')
-WORKLOAD_CONTROL_PLANE_ENDPOINT_IP=$(kubectl --kubeconfig=${KUBECONFIG} get "$(append_api_group ipaddresses)" "${WORKLOAD_IPADDRESS_NAME}" -o=jsonpath='{@.spec.address}')
+WORKLOAD_PLANE_IPCLAIM_NAME="ip-claim-$(openssl rand -hex 20)"
```
Comment (sbueringer) [inline hack/e2e.sh:124]: should we still add workload in some part of the name? Just makes it easier to distinguish later

File: hack/e2e.sh [hunk 4]
```diff
@@ -101,20 +101,29 @@ function kubectl_get_jsonpath() {
   fi
 }
 
+function claim_ip() {
+  IPCLAIM_NAME="$1"
+  export IPCLAIM_NAME
+  envsubst < "${REPO_ROOT}/hack/ipclaim-template.yaml" | kubectl --kubeconfig="${KUBECONFIG}" create -f - 1>&2
+  IPADDRESS_NAME=$(kubectl_get_jsonpath ipclaim "${IPCLAIM_NAME}" '{@.status.address.name}')
+  kubectl --kubeconfig="${KUBECONFIG}" get "$(append_api_group ipaddresses)" "${IPADDRESS_NAME}" -o=jsonpath='{@.spec.address}'
```
Comment (sbueringer) [inline hack/e2e.sh:109]: WDYT about getting rid of append_api_group? It just seems to obfuscate what we do

File: Makefile (hunk 2 above)
Comment (chrischdi) [reply Makefile]: nope, removed

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**